chain/chain/src/chain.rs

```diff
@@ -2070,13 +2070,29 @@ impl Chain {
 
         self.check_orphans(me, *block.hash(), block_processing_artifacts, apply_chunks_done_sender);
 
+        self.check_if_upgrade_needed(&block_hash);
+
         // Determine the block status of this block (whether it is a side fork and updates the chain head)
         // Block status is needed in Client::on_block_accepted_with_optional_chunk_produce to
         // decide to how to update the tx pool.
         let block_status = self.determine_status(new_head, prev_head);
         Ok(AcceptedBlock { hash: *block.hash(), status: block_status, provenance })
     }
 
+    fn check_if_upgrade_needed(&self, block_hash: &CryptoHash) {
```

Comment:
I like that it doesn't return an error, it wouldn't make sense to stop processing a block if this check failed.

Reply:
yeah, that's exactly what I have in mind :)